dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -228,6 +228,11 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
             String fileKey = file.getType().toString() + file.getAbsolutePath();
             SourceFile existingFile = existingFileMap.get(fileKey);
             if (existingFileMap.containsKey(fileKey)) {
+                // if the file is the primary descriptor and is about to change, then the ai topic sentence is dirty
+                if (file.getPath().equals(remoteVersion.getWorkflowPath()) && !existingFile.getContent().equals(file.getContent())) {
```

Comment:
Might be better to use the absolute path here, `file.getAbsolutePath().equals(...`, I think we're enforcing that `getWorkflowPath` is absolute now...

Reply:
:shrug: